### PR TITLE
Implemented a dark theme

### PR DESCRIPTION
**Pull request by emilyploszaj**
* Migrated most hard coded colors that matter to css variables
* Created a second set for a dark theme
* Set up the ability to theme switch, inferring from system theme

### CODE REVIEW

File: templates/search.hbs [hunk 1]
```diff
@@ -5,6 +5,8 @@
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
 
+	<script src="/static/js/main.js"></script>
```
Comment (Geometrically) [inline templates/search.hbs:8]: Could you move these to the bottom of the body, where the rest of the scripts are?

File: templates/mod-page.hbs [hunk 2]
```diff
@@ -5,6 +5,8 @@
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
 
+	<script src="/static/js/main.js"></script>
```
Comment (Geometrically) [inline templates/mod-page.hbs:8]: See other comment

File: templates/mod-create.hbs [hunk 3]
```diff
@@ -5,6 +5,8 @@
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
 
+	<script src="/static/js/main.js"></script>
```
Comment (Geometrically) [inline templates/mod-create.hbs:8]: See other comment

**Comment by emilyploszaj** [comment]
It's better for the script to be loaded before the css is so it think the theme has not changed and doesn't occasionally transition colors on page load

**Comment by Geometrically** [comment]
Makes sense.

**Pull request merged**